paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -229,6 +233,57 @@ internal class CustomerSheetViewModel @Inject constructor(
     }
 
     private fun onPrimaryButtonPressed() {
+        when (val currentViewState = viewState.value) {
+            is CustomerSheetViewState.AddPaymentMethod -> {
+                TODO()
+            }
+            is CustomerSheetViewState.SelectPaymentMethod -> {
+                when (val paymentSelection = currentViewState.paymentSelection) {
+                    PaymentSelection.GooglePay -> selectGooglePay()
+                    is PaymentSelection.Saved -> selectSavedPaymentMethod(paymentSelection)
+                    else -> error("$paymentSelection is not supported")
+                }
+            }
+            else -> error("${viewState.value} is not supported")
+        }
+    }
+
+    private fun selectSavedPaymentMethod(savedPaymentSelection: PaymentSelection.Saved) {
+        viewModelScope.launch {
+            customerAdapter.setSelectedPaymentOption(
+                savedPaymentSelection.toSavedSelection()?.toPaymentOption()
+            ).onFailure {
+                // TODO (jameswoo) Figure out what to do if payment option is unable to be persisted
+                updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
+                    it.copy(errorMessage = "Unable to save the selected payment option")
+                }
+            }.onSuccess {
+                val paymentMethod = savedPaymentSelection.paymentMethod
+                val paymentMethodId = paymentMethod.id
```

Comment:
While this technically _is_ nullable, it won’t be null in practice. I feel like we can `!!` this.